python/pyarrow/tests/test_feather.py

```diff
@@ -820,3 +826,20 @@ def test_feather_v017_experimental_compression_backward_compatibility(datadir):
     expected = pa.table({'a': range(5)})
     result = read_table(datadir / "v0.17.0.version.2-compression.lz4.feather")
     assert result.equals(expected)
+
+
+@pytest.mark.pandas
+def test_preserve_index_pandas(version):
+    data = {}
+    for i in range(4):
+        values = np.random.randint(0, 100, size=100)
+        data[i] = values
+
+    df = pd.DataFrame(data, index=data[0])
```

Comment:
```suggestion
    df = pd.DataFrame({'a': [1, 2, 3]}, index=['a', 'b', 'c'])
```

A DataFrame like that should also cover the behaviour, and makes the test code a bit simpler.